### PR TITLE
feat: added dark mode

### PR DESCRIPTION
**Pull request by Nishant-96**
This pr includes dark mode feature and some css changes for the same.

### CODE REVIEW

**Review by mohammadkhizerkhan — APPROVED** [review]
good naming convention for CSS variables
good to merge

File: src/components/navbar/navbar.jsx [hunk 1]
```diff
@@ -10,9 +10,8 @@ import LogoutIcon from "@mui/icons-material/Logout";
 import { Link, useNavigate } from "react-router-dom";
 import { useAuth } from "../../context/auth/auth-context";
 
-export function Navbar() {
-  //   will be removed state mangement for dark mode
-  const flager = true;
+export function Navbar({ changeTheme, theme }) {
+  console.log(theme);
```
Comment (znjs) [inline src/components/navbar/navbar.jsx]: can remove the console here

File: src/App.css [hunk 2]
```diff
@@ -6,8 +6,35 @@
...
+#dark-mode {
+  --nav-bg-color: var(--black-pearl);
+  --nav-color: var(--gray-white);
+  --nav-box-shadow: var(--blue);
+  --body-color: var(--white-two);
+  --side-nav-hover: var(--black-fade);
+  --note-input-shadow: 0px 0px 0.3rem var(--lightBlue);
+  --auth-wrapper-shadow: 0px 0px 0.9rem rgb(0 0 0 / 90%);
+  /* --gray: #5f6368;
+  --blue: #0063e5; */
```
Comment (znjs) [inline src/App.css]: can remove the commented code

**Review by znjs — APPROVED** [review]

**Comment by znjs** [comment]
Preview working as expected

**Pull request merged**